addons/point_of_sale/models/pos_order.py

```diff
@@ -659,7 +659,7 @@ def _prepare_aml_values_list_per_nature(self):
             if self.fiscal_position_id:
                 account = self.fiscal_position_id.map_account(account)
 
-            is_refund = line.qty * line.price_unit < 0
+            is_refund = -line.qty * line.price_unit < 0
```

Comment:
You are breaking everything with that. Let's start from the beginning:

The line of a pos order are already signed (+ for a sale, - for a refund (negative quantity)).
It's a refund if the line is negative so this line is completely correct, **remove your diff!**

Below (L671), you see `quantity=-line.qty,`. This is because for a sale, the product and tax lines need accounting journal items in credit.

When asking for an invoice after the closing of the POS entry, you enter `_create_misc_reversal_move` (L801).
In this method, we call `_prepare_aml_values_list_per_nature` and then we reverse the accounting amounts.
**Here is the error!**
The refund is not considered as a refund since the pos order is always the same.
Instead, add an extra `sign=1` parameter (named that way because there is one in the following versions because of Mexico).
Then:
`is_refund = line.qty * line.price_unit * sign < 0`
and
`quantity=-line.qty * sign,`
and you can remove the reverse of accounting amounts:
```py
aml_values['balance'] = -aml_values['balance']
aml_values['amount_currency'] = -aml_values['amount_currency']
```
:boom:

Reply:
@agr-odoo Forget everything in the comment above. Instead, let's force the `inverse_tax_tags` field as:
`inverse_tax_tags = not is_refund`

On the 2 loops below, force this value (L684 & L704):
```py
        # Create the tax lines
        for tax_line_vals in tax_results['tax_lines_to_add']:
            tax_rep = self.env['account.tax.repartition.line'].browse(tax_line_vals['tax_repartition_line_id'])
            amount_currency = tax_line_vals['tax_amount']
            balance = company_currency.round(amount_currency * rate)
            aml_vals_list_per_nature['tax'].append({
                'name': tax_rep.tax_id.name,
                'account_id': tax_line_vals['account_id'],
                'partner_id': tax_line_vals['partner_id'],
                'currency_id': tax_line_vals['currency_id'],
                'tax_repartition_line_id': tax_line_vals['tax_repartition_line_id'],
                'tax_ids': tax_line_vals['tax_ids'],
                'tax_tag_ids': tax_line_vals['tax_tag_ids'],
                'group_tax_id': None if tax_rep.tax_id.id == tax_line_vals['tax_id'] else tax_line_vals['tax_id'],
                'amount_currency': amount_currency,
                'balance': balance,
                'inverse_tax_tags': inverse_tax_tags, # <- HERE
            })
            total_amount_currency += amount_currency
            total_balance += balance
        # Create the aml values for order lines.
        for base_line_vals, update_base_line_vals in tax_results['base_lines_to_update']:
            order_line = base_line_vals['record']
            amount_currency = update_base_line_vals['price_subtotal']
            balance = company_currency.round(amount_currency * rate)
            aml_vals_list_per_nature['product'].append({
                'name': order_line.full_product_name,
                'account_id': base_line_vals['account'].id,
                'partner_id': base_line_vals['partner'].id,
                'currency_id': base_line_vals['currency'].id,
                'tax_ids': [(6, 0, base_line_vals['taxes'].ids)],
                'tax_tag_ids': update_base_line_vals['tax_tag_ids'],
                'amount_currency': amount_currency,
                'balance': balance,
                'inverse_tax_tags': inverse_tax_tags, # <- HERE
            })
            total_amount_currency += amount_currency
            total_balance += balance
```